var/spack/repos/builtin/packages/mimalloc/package.py

```diff
@@ -12,12 +12,45 @@ class Mimalloc(CMakePackage):
     homepage = "https://microsoft.github.io/mimalloc"
     url = "https://github.com/microsoft/mimalloc/archive/v0.0.0.tar.gz"
     git = "https://github.com/microsoft/mimalloc.git"
-    maintainers = ['msimberg']
+    maintainers = ["msimberg"]
 
-    version('dev-slice', branch='dev-slice')
-    version('dev', branch='dev')
-    version('master', branch='master')
-    version('2.0.6', sha256='9f05c94cc2b017ed13698834ac2a3567b6339a8bde27640df5a1581d49d05ce5')
-    version('1.7.6', sha256='d74f86ada2329016068bc5a243268f1f555edd620b6a7d6ce89295e7d6cf18da')
+    version("dev-slice", branch="dev-slice")
+    version("dev", branch="dev")
+    version("master", branch="master")
+    version("2.0.6", sha256="9f05c94cc2b017ed13698834ac2a3567b6339a8bde27640df5a1581d49d05ce5")
+    version("1.7.6", sha256="d74f86ada2329016068bc5a243268f1f555edd620b6a7d6ce89295e7d6cf18da")
 
-    depends_on('cmake@3.0:', type='build')
+    depends_on("cmake@3.0:", type="build")
+
+    mimalloc_options = {
+        "secure": (False, "Use full security mitigations (like guard pages, allocation randomization, double-free mitigation, and free-list corruption detection)", None),
+        "debug_full": (False, "Use full internal heap invariant checking in DEBUG mode (expensive)", None),
+        "padding": (True, "Enable padding to detect heap block overflow (used only in DEBUG mode)", None),
+        "override": (True, "Override the standard malloc interface (e.g. define entry points for malloc() etc)", None),
+        "xmalloc": (False, "Enable abort() call on memory allocation failure by default", None),
+        "show_errors": (False, "Show error and warning messages by default (only enabled by default in DEBUG mode)", None),
+        "use_cxx": (False, "Use the C++ compiler to compile the library (instead of the C compiler)", None),
+        "see_asm": (False, "Generate assembly files", None),
+        "osx_interpose": (True, "Use interpose to override standard malloc on macOS", "platform=darwin"),
+        "osx_zone": (True, "Use malloc zone to override standard malloc on macOS", "platform=darwin"),
```

Comment:
:shrug: I will see if I find something about it.

Reply:
As far as I understand they are recommended to be set together, but are not a must. This is what the mimalloc `CMakeLists.txt` has to say about it: https://github.com/microsoft/mimalloc/blob/f2712f4a8f038a7fb4df2790f4c3b7e3ed9e219b/CMakeLists.txt#L89-L101. Since there don't seem to be any hard errors or requirements on how they can be combined I wouldn't add any constraints here. But someone on macOS who knows more about what those options do may know better.